homeassistant/components/conversation/default_agent.py

```diff
@@ -561,6 +598,22 @@ def _get_or_load_intents(
                             ),
                             dict,
                         ):
+                            # Add metadata so we can identify custom sentences in the debugger
+                            custom_intents_dict = custom_sentences_yaml.get(
+                                "intents", {}
+                            )
+                            for intent_dict in custom_intents_dict.values():
+                                intent_data_list = intent_dict.get("data", [])
+                                for intent_data in intent_data_list:
```

Comment:
Seems expensive to do when just needed for debugging.

Reply:
I don't think it's very expensive; it only needs to loop through each intent and *block* of sentences. This is probably a few dozen items for heavy users (number of unique intents x number of sentence blocks).

Plus, it's only done once when the custom sentences are loaded.

Reply:
nevermind, you are right. This is only in loading, not when processing.